exporter/splunkhecexporter/metricdata_to_splunk.go

```diff
@@ -15,97 +15,134 @@
 package splunkhecexporter
 
 import (
-	"errors"
 	"fmt"
 	"math"
-	"strconv"
 
-	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
 	"go.opentelemetry.io/collector/consumer/pdata"
 	"go.opentelemetry.io/collector/translator/conventions"
-	"go.opentelemetry.io/collector/translator/internaldata"
 	"go.uber.org/zap"
-	"google.golang.org/protobuf/types/known/timestamppb"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/splunk"
 )
 
 const (
 	// unknownHostName is the default host name when no hostname label is passed.
 	unknownHostName = "unknown"
-	// separator for metric values.
-	separator = "."
 	// splunkMetricValue is the splunk metric value prefix.
 	splunkMetricValue = "metric_name"
-	// bucketSuffix is the bucket suffix for distribution buckets.
-	bucketSuffix = "bucket"
-	// quantileSuffix is the quantile suffix for summary quantiles.
-	quantileSuffix = "quantile"
 	// countSuffix is the count metric value suffix.
 	countSuffix = "count"
-	// sumOfSquaredDeviation is the sum of squared deviation metric value suffix.
-	sumOfSquaredDeviation = "sum_of_squared_deviation"
-)
-
-var (
-	// infinity bound dimension value is used on all histograms.
-	infinityBoundSFxDimValue = float64ToDimValue(math.Inf(1))
 )
 
 func metricDataToSplunk(logger *zap.Logger, data pdata.Metrics, config *Config) ([]*splunk.Event, int, error) {
-	ocmds := internaldata.MetricsToOC(data)
 	numDroppedTimeSeries := 0
-	_, numPoints := data.MetricAndDataPointCount()
-	splunkMetrics := make([]*splunk.Event, 0, numPoints)
-	for _, ocmd := range ocmds {
-		var host string
-		if ocmd.Resource != nil {
-			host = ocmd.Resource.Labels[conventions.AttributeHostHostname]
+	var splunkMetrics []*splunk.Event
+	for i := 0; i < data.ResourceMetrics().Len(); i++ {
+		rm := data.ResourceMetrics().At(i)
+
+		host := unknownHostName
+		source := config.Source
+		sourceType := config.SourceType
+		index := config.Index
+		if !rm.Resource().IsNil() {
+			if conventionHost, isSet := rm.Resource().Attributes().Get(conventions.AttributeHostHostname); isSet {
+				host = conventionHost.StringVal()
+			}
+			if sourceSet, isSet := rm.Resource().Attributes().Get(conventions.AttributeServiceName); isSet {
+				source = sourceSet.StringVal()
+			}
+			if sourcetypeSet, isSet := rm.Resource().Attributes().Get(splunk.SourcetypeLabel); isSet {
+				sourceType = sourcetypeSet.StringVal()
+			}
 		}
-		if host == "" {
-			host = unknownHostName
+		rm.InstrumentationLibraryMetrics().Len()
+		commonFields := map[string]interface{}{}
+		if !rm.Resource().IsNil() {
+			rm.Resource().Attributes().ForEach(func(k string, v pdata.AttributeValue) {
+				commonFields[k] = v.StringVal()
+			})
```

Comment:
Not all attributes are string, you may want to call `tracetranslator.AttributeValueToString`

Reply:
Do we need to convert non-string attribute values to string here or we want to preserve the type?